### ISSUE TITLE
Typeahead: Wikipedia example is not working at http://plnkr.co

### ISSUE THREAD

**Issue by goshoo** [ng-bootstrap/ng-bootstrap#1885]
Dear developers,

At first I want to tank you for providing such nice library for Angular 2+ and Bootstrap 4.

I'm  trying to use "Typeahead" component but I noticed that "Wikipedia" example didn't work on http://plnkr.co while the same example worked on your demo page.

I've never used this component so it'll be very useful for me to be able to see a working example.

Kind regards
Georgi

**Comment by volpejoaquin** [comment]
Any notice about that? I really need this functionality.

**Comment by goshoo** [comment]
Hi @volpejoaquin ,

I don't know why the example doesn't work at http://plnkr.co but the code works. Just do not forget to import all "rxjs/*" stuff at beginning of the example.

**Comment by bastienmoulia** [comment]
With rxjs@5.5 it's not working, but with rxjs@5.4 it work fine : http://plnkr.co/edit/a0y3cGYIESXimTKCOxBX?p=preview

**Comment by volpejoaquin** [comment]
@bastienmoulia you're right. Thanks for your quick response!

**Comment by moshmage** [comment]
The same happens with `typescript@2.4.2` (we jumped on ahead and are trying `angular@5-rc9` so we kinda need the typescript bump) the weird thing is that you wont get any console error (on plunkr, on local you'll get something like "debounceTime is not part of Observable" while building)

Not sure if this is caused by typescript, rxjs or a combination of wrong versions - though angular-cli requires Typescript^2.4.0 to work with AOT

I've also tried to use RxJS6 (next branch) to no avail. 

with `"rxjs": "^5.4.0",` and `"typescript": "~2.4.2"` using the basic example,
```
// component.html
<input id="typeahead-basic" name="typeahead" type="text" class="form-control"
                       [(ngModel)]="model" [ngbTypeahead]="search" [showHint]="true"/>
// component.ts
search = (text$: Observable<string>) => {
    console.log('stuff happened');
  }
```
will throw
```logtalk
ERROR TypeError: Cannot read property 'lift' of undefined
    at tapOperatorFunction (tap.js:52)
    at _do (do.js:48)
    at NgbTypeahead.ngOnInit (typeahead.js:73)
    at checkAndUpdateDirectiveInline (core.js:12094)
    at checkAndUpdateNodeInline (core.js:13597)
    at checkAndUpdateNode (core.js:13540)
    at debugCheckAndUpdateNode (core.js:14412)
    at debugCheckDirectivesFn (core.js:14353)
    at Object.View_BasicAddonConfigurationComponent_0._co [as updateDirectives] (BasicAddonConfigurationComponent.html:112)
    at Object.debugUpdateDirectives [as updateDirectives] (core.js:14338)
```

The following link _does not mimic the reported behaviour_ as it doesn't scream bloody murder on the console - though it doesn't even boot.

http://plnkr.co/edit/TJpNMLWgncJfFmST2unI?p=preview

**Comment by moshmage** [comment]
Ok. I was going to edit the post above but I felt like I'd change history so.. Here's how I "fixed" it:

I added `"noStrictGenericChecks": true,` to `tsconfig.json` and then issued `npm install --save rxjs@6.0.0-alpha.0`

I'm not sure why the previous RxJS@6.0.0-alpha.0 didn't work but I'm no longer having issues with `lift` though it still says `debounceTime` is not part of the `Observable` type.

**Comment by mfodor** [comment]
I've faced with similar error (ERROR TypeError: Cannot read property 'lift' of undefined).

For me it was because the search function has not returned anything however it should. There are no curly brackets ({}) in the example in the arrow function so it returns the value of the only expression. However, I've surrounded the function body with {} but not prefixed with `return`.

So here is my answer in 'code language':

Before: (mine, wrong)
```
search = (text$: Observable<string>) => {
    text$
      .debounceTime(300)
      ...
      .merge(this.hideSearchingWhenUnsubscribed);
}
```

After: (corrected; instead you can also use it without curly brackets and then you can skip the return keyword)
```
search = (text$: Observable<string>) => {
    return text$
      .debounceTime(300)
      ...
      .merge(this.hideSearchingWhenUnsubscribed);
};
```

**Comment by fixbugsnow** [comment]
Thanks mfodor! Just adding "return" before "text$" resolved my "Cannot read property 'lift' of undefined" error,

Using:
@angular/... : "^5.2.0"
@ng-bootstrap/ng-bootstrap: "^1.0.0-beta.9"
rxjs: "5.5.6"
typescript: "~2.5.3"

**Comment by anilvangari2005** [comment]
Solution by @mfodor works. I was missing return statement. Pay attention to the function syntax.
